### PR TITLE
spirv: don't emit invalid debuginfo for buffer references

### PR DESCRIPTION
**Pull request by arcady-lunarg**
Currently no debug info is emitted for buffer reference types, which resulted in the SPIR-V backend code asserting when trying to emit the debug info for struct member that had such a type. Instead, the code now skips such struct members. Full debug info for buffer references may require forward references in the debug info instructions, which is currently prohibited by the spec.

### CODE REVIEW

File: SPIRV/SpvBuilder.cpp [hunk 1]
```diff
@@ -1029,7 +1029,10 @@ Id Builder::makeCompositeDebugType(std::vector<Id> const& memberTypes, char cons
     for(auto const memberType : memberTypes) {
         assert(debugTypeLocs.find(memberType) != debugTypeLocs.end());
 
-        memberDebugTypes.emplace_back(makeMemberDebugType(memberType, debugTypeLocs[memberType]));
+        // There _should_ be debug types for all the member types but currently buffer references
+        // do not have member debug info generated.
+        if (debugId[memberType])
```
Comment (jeremy-lunarg) [inline SPIRV/SpvBuilder.cpp:1034]: This could filter a larger set than just buffer references, but I think it's okay for now.

**Review by jeremy-lunarg — APPROVED** [review]
Okay for now. Hopefully this gets better when we get the new extensions.

File: SPIRV/SpvBuilder.cpp (hunk 1 above)
Comment (arcady-lunarg) [reply SPIRV/SpvBuilder.cpp:1034]: If we've gotten this far, the options are emit invalid debug info (with zero Ids) or not emit any debuginfo at all, and I think the latter is preferable.

**Review by ncesario-lunarg — APPROVED** [review]

File: SPIRV/SpvBuilder.cpp (hunk 1 above)
Comment (jeremy-lunarg) [reply SPIRV/SpvBuilder.cpp:1034]: I agree, but it would be nice to know where the other holes might be.

**Pull request merged**